dessia_common/workflow/core.py

```diff
@@ -434,22 +467,36 @@ def __deepcopy__(self, memo=None):
                                    imposed_variable_values=imposed_variable_values, name=self.name)
         return copied_workflow
 
-    def copy_pipe(self, pipe: Pipe, copied_workflow: 'Workflow') -> Pipe:
-        """
-        Copy a pipe to another workflow
-        """
-        upstream_index = self.variable_indices(pipe.input_variable)
+    def copy_pipes(self, copied_workflow: 'Workflow') -> List[Pipe]:
```

Comment:
In this case, we could set a `copy_pipe` function that would be called in the `for pipe in self.pipes` loop and thus guarantee retro-compatibility, at least for the standard pipes.

If it finds an nbv at upstream, it could raise an error, that copying pipes related to nbs isn't trivial